src/cloudsc_cuda/CMakeLists.txt

```diff
@@ -0,0 +1,181 @@
+            cloudsc/cloudsc_c.cu
+            cloudsc/cloudsc_driver.h
+            cloudsc/cloudsc_driver.cu
+            cloudsc/cloudsc_validate.h
+            cloudsc/cloudsc_validate.cu
+            cloudsc/mycpu.h
+            cloudsc/mycpu.cu
+        PUBLIC_INCLUDES
+            $<INSTALL_INTERFACE:include>
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
+        PUBLIC_LIBS
+            Serialbox::Serialbox_C
+            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+    )
+
+    target_include_directories(
+      dwarf-cloudsc-cuda-lib
+      PUBLIC 
+      ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+    )
+    target_compile_options(dwarf-cloudsc-cuda-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
+	    -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
+	       
+    set_target_properties( dwarf-cloudsc-cuda-lib PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+    
+    ecbuild_add_executable(
+        TARGET  dwarf-cloudsc-cuda
+        SOURCES dwarf_cloudsc.cpp
+        LIBS dwarf-cloudsc-cuda-lib
+    )
+
+    target_link_libraries(dwarf-cloudsc-cuda dwarf-cloudsc-cuda-lib)
+
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-cuda-serial
+        COMMAND bin/dwarf-cloudsc-cuda
+        ARGS 1 1000 128
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 1
+    )
+    ###
+
+    ###### SCC-CUDA-HOIST ####
+    ecbuild_add_library(
+        TARGET dwarf-cloudsc-cuda-hoist-lib
+        INSTALL_HEADERS LISTED
+        SOURCES
+            cloudsc/yoecldp_c.h
+            cloudsc/load_state.h
+            cloudsc/load_state.cu
+            cloudsc/cloudsc_c_hoist.h
+            cloudsc/cloudsc_c_hoist.cu
+            cloudsc/cloudsc_driver_hoist.h
+            cloudsc/cloudsc_driver_hoist.cu
+            cloudsc/cloudsc_validate.h
+            cloudsc/cloudsc_validate.cu
+            cloudsc/mycpu.h
+            cloudsc/mycpu.cu
+        PUBLIC_INCLUDES
+            $<INSTALL_INTERFACE:include>
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
+        PUBLIC_LIBS
+            Serialbox::Serialbox_C
+            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+    )
+
+    target_include_directories(
+      dwarf-cloudsc-cuda-hoist-lib
+      PUBLIC
+      ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+    )
+    target_compile_options(dwarf-cloudsc-cuda-hoist-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
+                       -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
+
+    set_target_properties( dwarf-cloudsc-cuda-hoist-lib PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+
+    ecbuild_add_executable(
+        TARGET  dwarf-cloudsc-cuda-hoist
+        SOURCES dwarf_cloudsc.cpp
+        LIBS dwarf-cloudsc-cuda-hoist-lib
+    )
+
+    target_link_libraries(dwarf-cloudsc-cuda-hoist dwarf-cloudsc-cuda-hoist-lib)
+    
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-cuda-hoist-serial
+        COMMAND bin/dwarf-cloudsc-hoist-caching
```

Comment:
This test doesn't work, I suspect because the command is wrong. Should be bin/dwarf-cloudsc-cuda-hoist